drivers/pci/pci_shell.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell.h>
+#include <init.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <logging/log.h>
+#include <pci/pci.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+
+LOG_MODULE_REGISTER(lspci_shell);
+
+static void list_devices(const struct shell *shell,
+			 struct pci_dev_info *dev_info)
+{
+
+	shell_fprintf(shell, SHELL_NORMAL,
+		      "%u:%u %X:%X class: 0x%X, %u, %u, %s, "
+		      "addrs: 0x%X-0x%X, IRQ %d\n",
+		      dev_info->bus,
+		      dev_info->dev,
+		      dev_info->vendor_id,
+		      dev_info->device_id,
+		      dev_info->class_type,
+		      dev_info->function,
+		      dev_info->bar,
+		      (dev_info->mem_type == BAR_SPACE_MEM) ? "MEM" : "I/O",
+		      (u32_t)dev_info->addr,
+		      (u32_t)(dev_info->addr + dev_info->size - 1),
+		      dev_info->irq);
+}
+
+static int cmd_lspci(const struct shell *shell, size_t argc, char **argv)
+{
+	struct pci_dev_info info = {
+		.function = PCI_FUNCTION_ANY,
+		.bar = PCI_BAR_ANY,
+	};
+
```

Comment:
You should implement here either:

```
int err = shell_cmd_precheck(shell, (argc == 1), NULL, 0);

if (err) {
        return err;
}
```

or:

```
if (hell_help_requested(shell) {
        shell_help_print(shell, NULL, 0);
        return;
}
```

Otherwise lspci -h and lspci --help won't work.